indexers/vector/SptagIndexer/tests/test_sptagindexer.py

```diff
@@ -75,18 +75,18 @@ def test_sptag_indexer_known(metas):
     with BaseIndexer.load(save_abspath) as indexer:
         assert isinstance(indexer, SptagIndexer)
         idx, dist = indexer.query(queries, top_k=2)
-        np.testing.assert_equal(idx, np.array([[4, 5], [5, 4], [6, 5], [7, 6]]))
+        np.testing.assert_equal(idx.astype(str), np.array([[4.0, 5.0], [5.0, 4.0], [6.0, 5.0], [7.0, 6.0]]).astype(str))
```

Comment:
But why do you need to convert to `str`? These should be strings from the return

Reply:
If I don't convert `lhs` (`idx`) to `str` explicitly, I get:
```
          HubIO@21[I]:E           Mismatched elements: 1 / 1 (100%)
          HubIO@21[I]:E            x: array([[4, 5],
          HubIO@21[I]:E                  [5, 4],
          HubIO@21[I]:E                  [6, 5],
          HubIO@21[I]:E                  [7, 6]])
          HubIO@21[I]:E            y: array([['4.0', '5.0'],
          HubIO@21[I]:E                  ['5.0', '4.0'],
          HubIO@21[I]:E                  ['6.0', '5.0'],
          HubIO@21[I]:E                  ['7.0', '6.0']], dtype='<U32')
```
As `unicode` comparison with `numeric` datatypes won't returned equal

Reply:
Yes, I understand. 

The problem is that SptagIndexer behaves differently. See Annoy here : https://github.com/jina-ai/jina-hub/blob/998a881b80841b47ba170c2f6cd1496a7aeb1b6d/indexers/vector/AnnoyIndexer/tests/test_annoyindexer.py#L100

You can see that in this case we are only converting to `str` the right-hand element in the comparison. That's what I would expect here as well. 

@JoanFM Any ideas why this is happening?

Reply:
exactly, u should not need to cast `idx`

Reply:
also u should not have it as `float`